File: scrape-movie-service/models/movie.go
package models

import (
	"github.com/jinzhu/gorm"
)

type Movie struct {
	Response   string
	Type       string
	Title      string
	Year       string
	Rated      string
	Released   string
	Runtime    string
	Genre      string
	Director   string
	Writer     string
	Actors     string
	Plot       string
	Language   string
	Country    string
	Poster     string
	ImdbRating string
	Metascore  string
	ImdbVotes  string
	ImdbID     string
	Error      string
}

type MoviePage struct {
	Movies []MovieDb `json:"movies"`
	// Page
	// SortType
	// SortDirection
	// ...
}

type MovieDb struct {
	gorm.Model   `json:"-"`
	Type         string `gorm:"not null;" json:"type"`
	Title        string `gorm:"not null;" json:"title"`
	Year         string `gorm:"not null;" json:"year"`
	Rated        string `gorm:"not null;" json:"rated"`
	Released     string `gorm:"not null;" json:"released"`
	Runtime      string `gorm:"not null;" json:"runtime"`
	Genre        string `gorm:"not null;" json:"genre"`
	Director     string `gorm:"not null;" json:"director"`
	Writer       string `gorm:"not null;" json:"writer"`
	Actors       string `gorm:"not null;" json:"actors"`
	Plot         string `gorm:"not null;" json:"plot"`
	Language     string `gorm:"not null;" json:"language"`
	Country      string `gorm:"not null;" json:"country"`
	Poster       string `gorm:"not null;" json:"poster"`
	ImdbRating   string `gorm:"not null;" json:"imdbRating"`
	Metascore    string `gorm:"not null;" json:"metascore"`
	ImdbVotes    string `gorm:"not null;" json:"imdbVotes"`
	ImdbID       string `gorm:"not null;" json:"imdbID"`
	TorrentLinks string `json:"torrentLinks"`
	TitleLinks   string `json:"titleLinks"`
}

type MovieUpdateDTO struct {
	ImdbLink     string `json:"imdbLink"`
	RottenLink   string `json:"rottenLink"`
	TorrentLinks string `json:"torrentLinks"`
	TitleLinks   string `json:"titleLinks"`
}

func UpdateMovieData(original *MovieDb, new *Movie) {
	original.Title = new.Title
	original.ImdbID = new.ImdbID
	original.Language = new.Language
	original.ImdbVotes = new.ImdbVotes
	original.Metascore = new.Metascore
	original.ImdbRating = new.ImdbRating
	original.Poster = new.Poster
	original.Country = new.Country
	original.Plot = new.Plot
	original.Actors = new.Actors
	original.Writer = new.Writer
	original.Director = new.Director
	original.Genre = new.Genre
	original.Runtime = new.Runtime
	original.Released = new.Released
	original.Rated = new.Rated
	original.Year = new.Year
	original.Type = new.Type
}

File: gleficu/src/main/java/com/sbnz/gleficu/model/movie/Movie.java
package com.sbnz.gleficu.model.movie;

import com.sbnz.gleficu.model.BaseEntity;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.OffsetDateTime;

@Getter @Setter @NoArgsConstructor @AllArgsConstructor
@Table(name = "movie_dbs")
@Entity
public class Movie extends BaseEntity {
    @Column(name = "created_at", nullable = true)
    private OffsetDateTime createdAt;

    @Column(name = "updated_at", nullable = true)
    private OffsetDateTime updatedAt;

    @Column(name = "deleted_at", nullable = true)
    private OffsetDateTime deletedAt;

    @Column(name="type")
    private String type;

    @Column(name="title")
    private String title;

    @Column(name="year")
    private String year;

    @Column(name="rated")
    private String rated;

    @Column(name="released")
    private String released;

    @Column(name="runtime")
    private String runtime;

    @Column(name="genre")
    private String genre;

    @Column(name="director")
    private String director;

    @Column(name="writer")
    private String writer;

    @Column(name="actors")
    private String actors;

    @Column(name="plot")
    private String plot;

    @Column(name="language")
    private String language;

    @Column(name="country")
    private String country;

    @Column(name="poster")
    private String poster;

    @Column(name="imdb_rating")
    private String imdbRating;

    @Column(name="metascore")
    private String metascore;

    @Column(name="imdb_votes")
    private String imdbVotes;

    @Column(name="imdb_id")
    private String imdbID;

    @Column(name="torrent_links")
    private String torrentLinks;

    @Column(name="title_links")
    private String titleLinks;
}
